src/components/SideBar/index.tsx

```diff
@@ -0,0 +1,169 @@
+'use client'
+
+import Link from "next/link";
+import Image from "next/image";
+import { usePathname, useRouter, useSearchParams } from 'next/navigation'
+import { useState } from "react";
+import {
+  ArrowRightFromLine,
+  DiscAlbum,
+  Flag,
+  Gauge,
+  Globe,
+  Guitar,
+  Home,
+  Library,
+  Moon,
+  Settings,
+  Sun,
+  User,
+  Users
+} from "lucide-react";
+import { parseClassName } from "@/utils/helpers";
+import SideBarHeader from "./SideBarHeader";
+import styles from "./sidebar.module.css";
+
+const navigation = [
+  {
+    links: [
+      {
+        href: "/home",
+        name: "Home",
+        icon: Home
+      },
+      {
+        href: "/community",
+        name: "Community",
+        icon: Globe,
+      },
+      {
+        href: "/dashboard",
+        name: "Dashboard",
+        icon: Gauge,
+      }
+    ]
+  },
+  {
+    separator: "Leaderboards",
+    links: [
+      {
+        href: "/player",
+        name: "Player",
+        icon: User
+      },
+      {
+        href: "/band",
+        name: "Band",
+        icon: Users,
+      },
+      {
+        href: "/instrument",
+        name: "Instrument",
+        icon: Guitar,
+      },
+      {
+        href: "/country",
+        name: "Country",
+        icon: Flag,
+      },
+      {
+        href: "/setlist",
+        name: "Setlist",
+        icon: Library,
+      },
+      {
+        href: "/source",
+        name: "Source",
+        icon: DiscAlbum,
+      }
+    ]
+  },
+  {
+    separator: true,
+    links: [
+      {
+        href: "/settings",
+        name: "Settings",
+        icon: Settings
+      }
+    ]
+  }
+]
+
+const languages = [
+  { name: "English", acronym: "en"},
+  { name: "Português", acronym: "pt"}
+]
+
+const release = {
+  image: "https://news.yarg.in/images/banners/v0-12-release.webp",
+  target: "#!",
+  text: "New release 0.12" 
+}
+
+const SideBar: React.FC = () => {
+  const router = useRouter();
+  const query = useSearchParams();
+  const isClose = query.has('closed')
+  const [theme, setTheme] = useState<'dark'|'light'>('dark')
+  const pathname = usePathname()
+
+  const handleTheme = () => setTheme((current) => current === 'dark' ? 'light': 'dark')
+
+  const handleToggle = () => 
+    router.push(`${pathname}${!isClose ? "?closed" : "?"}`)
+  
+  return (
+    <div className={styles.sidebar} aria-expanded={!isClose}>
+      {!isClose ? <SideBarHeader.Open tag="v0.12.0" onClose={handleToggle}/> : <SideBarHeader.Close />}
+      <div className={styles.navigation}>
+        {isClose && (
+          <button type="button" className={styles.link} onClick={handleToggle}>
+            <ArrowRightFromLine className={styles.icon} />
+          </button>
+        )}
+        {navigation?.map(({separator, links}, index) => (
+          <section key={index} className={styles.section}>
```

Comment:
Usar index não é uma boa prática no React, na reconciliação o index é uma chave muito fraca de comparação e pode levar a bugs. 

Digamos que por algum motivo aplicamos um sort ou mudamos um elemento de lugar, como o React olha apenas para o index os valores continuam os mesmos e por mais que o elemento referente a aquele index seja diferente para o react continua sendo o mesmo elemento e ele não é levado em conta como mudança no algoritmo reconciliação e seus elementos não vão ser apresentados na ordem que o sort realizou, ele vai ignorar essa mudança ou pode gerar itens duplicados.

Esse é um dos motivos de ser tão importante que nas keys tenhamos chaves realmente únicas que possam identificar diferentes elementos individualmente. Nesse caso como vc não necessariamente tem uma chave string talvez seja uma boa opção gerar um randomUUID da lib crypto ou você pode gerar um timestamp em conjunto com o JSON stringnify do objeto que ta percorrendo, dificilmente terá conflito com a ultima opção mas creio q a primeira seja mais recomendada.

Vou deixar aqui o link da documentação com esse problema em específico e porque ele é importante

[React rendering lists](https://react.dev/learn/rendering-lists#keeping-list-items-in-order-with-key)

Reply:
Aprendi algo novo hoje, obrigado!